modules/system/common/hardware/powersave/power-benchmark.sh

```
@@ -0,0 +1,247 @@
+    for bat_path in "${battery_paths[@]}"; do
+        if [[ -f "${bat_path}/power_now" ]]; then
+            power_uw=$(cat "${bat_path}/power_now" 2>/dev/null || echo 0)
+            [[ $power_uw -gt 0 ]] && break
+        fi
+    done
+    
+    # Method 2: Calculate from current and voltage
+    if [[ $power_uw -eq 0 ]]; then
+        for bat_path in "${battery_paths[@]}"; do
+            if [[ -f "${bat_path}/current_now" && -f "${bat_path}/voltage_now" ]]; then
+                local current_ua=$(cat "${bat_path}/current_now" 2>/dev/null || echo 0)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)